packages/flutter/lib/src/rendering/proxy_sliver.dart

```diff
@@ -399,3 +400,39 @@ class RenderSliverAnimatedOpacity extends RenderProxySliver with RenderAnimatedO
     child = sliver;
   }
 }
+
+/// Applies a cross-axis constraint to its sliver child.
+///
+/// This render object takes an [maxExtent] parameter and uses the smaller of
+/// [maxExtent] and the parent's [SliverConstraints.crossAxisExtent] as the
+/// cross axis extent of the [SliverConstraints] passed to the sliver child.
+class RenderSliverConstrainedCrossAxis extends RenderProxySliver {
+  /// Creates a render object that constrains the cross axis extent of its sliver child.
+  ///
+  /// The [maxExtent] parameter must not be null and must be between 0 and the
+  /// parent's cross axis extent.
```

Comment:
This needs to be updated since it does not need to be less than the parent's cross axis extent.